controllers/apps/v1beta4/rebalance_controller_test.go

```diff
@@ -63,3 +69,197 @@ func TestGetRebalanceStatus(t *testing.T) {
+			EmqxNodes: []appsv1beta4.EmqxNode{
+				{
+					Node: "emqx-ee@emqx-ee-0.emqx-ee-headless.default.svc.cluster.local",
+				},
+			},
+		},
+	}
+
+	emqxNodeName := "emqx-ee@emqx-ee-0.emqx-ee-headless.default.svc.cluster.local"
+
+	t.Run("check requestAPI args", func(t *testing.T) {
+		startPath := fmt.Sprintf("api/v4/load_rebalance/%s/start", emqxNodeName)
+		f.requestAPI = func(method, path string, body []byte) (resp *http.Response, respBody []byte, err error) {
+			assert.Equal(t, "POST", method)
+			assert.Equal(t, startPath, path)
+			assert.Equal(t, getRequestBody(rebalance, []string{emqxNodeName}), body)
+			resp = &http.Response{StatusCode: http.StatusOK}
+			respBody = []byte(`{"data":[],"code":0}`)
+			err = nil
+			return
+		}
+
+		err := startRebalance(f, rebalance, emqx, emqxNodeName)
+		assert.Nil(t, err)
+	})
+
+	t.Run("check requestAPI return err", func(t *testing.T) {
+		f.requestAPI = func(method, path string, body []byte) (resp *http.Response, respBody []byte, err error) {
+			return &http.Response{StatusCode: http.StatusBadRequest}, nil, nil
+		}
+
+		err := startRebalance(f, rebalance, emqx, emqxNodeName)
+		assert.ErrorContains(t, err, "request api failed")
+	})
+
+	t.Run("check requestAPI start rebalance err", func(t *testing.T) {
+		f.requestAPI = func(method, path string, body []byte) (resp *http.Response, respBody []byte, err error) {
+			resp = &http.Response{StatusCode: http.StatusOK}
+			respBody = []byte(`{"message":[\"nothing_to_balance\"],"code":400}`)
+			err = emperror.New("[\"nothing_to_balance\"]")
+			return
+		}
+		err := startRebalance(f, rebalance, emqx, emqxNodeName)
+		assert.ErrorContains(t, err, "nothing_to_balance")
+	})
+
+	t.Run("check requestAPI start rebalance success", func(t *testing.T) {
+		f.requestAPI = func(method, path string, body []byte) (resp *http.Response, respBody []byte, err error) {
+			return &http.Response{StatusCode: http.StatusOK}, []byte(`{"data":[],"code":0}`), nil
+		}
+		err := startRebalance(f, rebalance, emqx, emqxNodeName)
+		assert.Nil(t, err)
+	})
+
+	t.Run("check requestAPI return error", func(t *testing.T) {
+		f.requestAPI = func(method, path string, body []byte) (resp *http.Response, respBody []byte, err error) {
+			return nil, nil, errors.New("fake error")
+		}
+		err := startRebalance(f, rebalance, emqx, emqxNodeName)
+		assert.Error(t, err, "fake error")
+	})
+}
+
+func TestStopRebalance(t *testing.T) {
+	f := &fakePW{}
+	rebalance := &appsv1beta4.Rebalance{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "rebalance-test",
+		},
+		Spec: appsv1beta4.RebalanceSpec{
+			InstanceName: "emqx-ee",
+			RebalanceStrategy: appsv1beta4.RebalanceStrategy{
+				ConnEvictRate:    5,
+				SessEvictRate:    5,
+				WaitHealthCheck:  10,
+				WaitTakeover:     10,
+				AbsSessThreshold: 10,
+				AbsConnThreshold: 10,
+				RelConnThreshold: "1.1",
+				RelSessThreshold: "1.1",
+			},
+		},
+		Status: appsv1beta4.RebalanceStatus{
+			Phase: "Processing",
```

Comment:
This line is needless

Reply:
> This line is needless

why？ To stop rebalance, you need to confirm whether rebalance is in progress